templates/esm/typescript_gapic/esm/src/$version/$service_client.ts.njk

```diff
@@ -607,6 +608,9 @@ export class {{ service.name }}Client {
 {%- endif %}
       ]>|void {
     request = request || {};
+    {% for field in method.autoPopulatedFields %}if (!request.{{ field.toCamelCase() }}) {
+      request.{{ field.toCamelCase() }} = crypto.randomUUID();
```

Comment:
To make it compatible with browser, we have two options: either use `uuid` module instead of crypto (which will force us make `uuid` a dependency for all libraries), or - probably better - export a `createUUID()` function from gax, and have it there instead.